core/math/math_funcs.cpp

```diff
@@ -67,17 +67,19 @@ int Math::step_decimals(double p_step) {
 		0.000009999,
 		0.0000009999,
 		0.00000009999,
-		0.000000009999
+		0.000000009999,
+		0.0000000009999
 	};
 
-	double as = Math::abs(p_step);
+	double abs = Math::abs(p_step);
+	double decs = abs - (int)abs; // Strip away integer part
 	for (int i = 0; i < maxn; i++) {
-		if (as >= sd[i]) {
+		if (decs >= sd[i]) {
 			return i;
 		}
 	}
 
-	return maxn;
+	return 0;
```

Comment:
Should it return 0 if it couldn't find any significant digit before running out of things to compare with?

Maybe it could return `-1` instead, and this would be documented as a failure to process the given number.

Reply:
0 is what I would expect, but it's fine if we change it to -1, if that's desired.

Reply:
Nevermind, I get your point. If it can't be matched in the array, it's beyond the precision that we support, and therefore it *is* 0, i.e. `decimals(1.1)` and `decimals(1.100000000013)` should give the same result.